src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -381,7 +384,7 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                             (interfaceGenericType == typeof(ICollection<>) || interfaceGenericType == typeof(IList<>)))
                         {
                             // For ICollection<T> and IList<T> we bind them to mutable List<T> type.
-                            Type genericType = typeof(List<>).MakeGenericType(type.GenericTypeArguments[0]);
+                            Type genericType = typeof(List<>).MakeGenericType(type.GenericTypeArguments);
```

Comment:
> Type genericType = typeof(List<>).MakeGenericType(type.GenericTypeArguments); [](http://example.com/codeflow?start=27&length=78)

why this change? this doesn't look right to me. `type.GenericTypeArguments` will return array of the type and not just the type itself.

Reply:
It will return an array of type arguments for that type (with a single element) and `MakeGenericType` takes an array of type arguments (using `params` arrays), so this just avoids an array allocation here.